### PR TITLE
Add double vote detection to spanner

### PR DESCRIPTION
**Pull request by 0xKiwi**
Part of #4836 

This PR adds support for double vote detection to the spanner for catching double votes reliably.

This PR changes the span structure of [2]uint16 to [3]uint16, adding a 2 byte bloom filter encoded as a uint16 for easy storage. 

It marks the target epoch of the spanner with a bloom filter representing the attestation data root submitted. If we receive another attestation from that validator for that epoch we can reliably check it with the bloom filter to ensure it is a unique data root efficiently.

### CODE REVIEW

File: slasher/detection/attestations/spanner.go [hunk 1]
```diff
@@ -149,13 +180,45 @@ func (s *SpanDetector) UpdateSpans(ctx context.Context, att *ethpb.IndexedAttest
...
+			return errors.Wrap(err, "failed to update attestation filter")
+		}
 		// Update min and max spans.
 		s.updateMinSpan(source, target, valIdx)
 		s.updateMaxSpan(source, target, valIdx)
 	}
 	return nil
 }
 
+func (s *SpanDetector) markAttFilter(attData *ethpb.AttestationData, valIdx uint64) error {
```
Comment (terencechain) [inline slasher/detection/attestations/spanner.go]: Can you please add a few comments in `markAttFilter`? 
I can imagine it's quite hard for an outsider to figure out what this function does

File: slasher/detection/attestations/spanner.go [hunk 2]
```diff
@@ -92,12 +97,38 @@ func (s *SpanDetector) DetectSlashingForValidator(
 			}, nil
 		}
 	}
+
+	if sp := s.spans[targetEpoch%numSpans]; sp != nil {
```
Comment (terencechain) [inline slasher/detection/attestations/spanner.go]: `sp, ok := s.spans[targetEpoch%numSpans]`

Use `if !ok` ?

Comment (0xKiwi) [reply slasher/detection/attestations/spanner.go]: This is an array, not a map

File: slasher/detection/attestations/spanner.go (hunk 1 above)
Comment (0xKiwi) [reply slasher/detection/attestations/spanner.go]: Sure thing

File: slasher/detection/attestations/spanner.go [hunk 3]
```diff
@@ -149,13 +180,52 @@ func (s *SpanDetector) UpdateSpans(ctx context.Context, att *ethpb.IndexedAttest
...
+			return errors.Wrap(err, "failed to update attestation filter")
+		}
 		// Update min and max spans.
 		s.updateMinSpan(source, target, valIdx)
 		s.updateMaxSpan(source, target, valIdx)
 	}
 	return nil
 }
 
+// markAttFilter sets third uint16 in the target epochs span to a bloom filter
```
Comment (0xKiwi) [inline slasher/detection/attestations/spanner.go]:
```suggestion
// markAttFilter sets the third uint16 in the target epochs span to a bloom filter
```

**Pull request merged**